Basic/2DSpaceShooter/Assets/Scripts/ShipControl.cs

```diff
@@ -180,11 +180,16 @@ public void TakeDamage(int amount)
 
         if (Health.Value <= 0)
         {
-            Health.Value = 0;
```

Comment:
Why was this line deleted?

Reply:
Since Health is a NetworkVariable with events attached to its OnValueChanged, there is something that happens in between: `SetHealthBarValue` gets called.

If `SetHealthBarValue` had some code like:

```
if (Health.Value == 0)
{ 
     Debug.Log("You're dead"); 
     //do other stuff
}
```

The change would break the game.

Reply:
I see, you are right it is important to check for that. It does not seem like this is the case here, but I will still add the line back, as it is good visual feedback for the player/user to have the health at 0 before it gets reset.